### PR TITLE
fix(file-explorer): allow filtered folder toggles

### LINKED ISSUE #5715: [Bug]: File explorer expand/collapse does not work in name search mode
### Operating system

macOS

### Orca version

1.4.77

### Details

Short summary:                                                                                            
When the file explorer is in name search mode, folder expand/collapse controls in the visual explorer do not respond.

I opened the file explorer, entered name search mode, then tried to expand or collapse a visible folder using the disclosure arrow/caret. Nothing happened. Outside name search mode, expand/collapse works normally.

How can we reproduce it?
1. Open Orca.
2. Open a workspace/worktree with nested folders.
3. Open the file explorer.
4. Enter name search mode.
5. Search for a file/folder name so matching items are visible.
6. Try to expand or collapse a visible folder in the visual explorer.
7. Exit name search mode and try the same expand/collapse action again.

Expected behavior:
Folders in the visual explorer should still expand/collapse while name search mode is active, or the UI should make it clear that tree expansion is disabled during search.

Actual behavior:
Expand/collapse does not work while name search mode is active.

Anything else that might help:
This seems minor but makes it hard to inspect nearby files while filtering by name.

### PR DESCRIPTION
**Pull request by bennewell35**
## Summary
- Allow directory rows to expand and collapse while file-name filtering is active.
- Keep filtered collapse state transient so normal persisted file explorer expansion is unchanged.
- Add regression coverage for the filtered projection and toggle helper.

Closes #5715

## Testing
- `pnpm exec vitest run --config config/vitest.config.ts src/renderer/src/components/right-sidebar/useFileExplorerVisibleRowProjection.test.ts src/renderer/src/components/right-sidebar/FileExplorer.test.tsx src/renderer/src/components/right-sidebar/useFileExplorerHandlers.test.ts src/renderer/src/components/right-sidebar/file-explorer-keyboard-navigation.test.ts`
- `pnpm run lint`
- `pnpm run typecheck`
- `pnpm test`
- `pnpm run build:desktop` (passes; local `/usr/local/bin/orca-dev` symlink permission warning and existing Vite chunk/dynamic-import warnings)

## Notes
- No platform-specific shortcuts, path handling, SSH behavior, auth, or trust boundary changes.

---

<a href="https://stagereview.app/stablyai/orca/pull/5977">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

### CODE REVIEW

**Comment by ghost** [comment]
Ready to review this PR? Stage has broken it down into 5 individual chapters for you:

| | Title |
|---|-------|
| 1 | [Add helpers for filtered folder toggles](https://stagereview.app/stablyai/orca/pull/5977/chapters/1) |
| 2 | [Support collapsed paths in filtered projections](https://stagereview.app/stablyai/orca/pull/5977/chapters/2) |
| 3 | [Wire collapsed paths through projection hook](https://stagereview.app/stablyai/orca/pull/5977/chapters/3) |
| 4 | [Enable folder toggling in FileExplorer UI](https://stagereview.app/stablyai/orca/pull/5977/chapters/4) |
| 5 | [Update keyboard navigation for filtered toggles](https://stagereview.app/stablyai/orca/pull/5977/chapters/5) |

<a href="https://stagereview.app/stablyai/orca/pull/5977">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

<sub>Chapters generated by <a href="https://stagereview.app">Stage</a> for commit 18e1dfc7034fe05f120f4922780a83f22480597f on Jun 22, 2026 9:02pm UTC.</sub>

File: src/renderer/src/components/right-sidebar/FileExplorer.tsx [hunk 1]
```diff
@@ -125,6 +131,9 @@ function FileExplorerFiles(): React.JSX.Element {
     [nameFilterQuery]
   )
   const hasNameFilter = isFilesViewActive && hasNameFilterQuery
+  useEffect(() => {
+    setNameFilterCollapsedPaths((current) => (current.size > 0 ? new Set() : current))
+  }, [hasNameFilter, nameFilterQuery])
```
[the remark this answers is not in the document]
Comment (bennewell35) [reply src/renderer/src/components/right-sidebar/FileExplorer.tsx]: Addressed in 01ccd538e by clearing the transient filtered collapse state only when name filtering is deactivated, so query edits preserve the active filtered collapse state.

**Review by Jinwoo-H — APPROVED** [review]
Thanks, this is a good focused fix. I found and patched one edge case during review: filtered drag-hover auto-expand now uses the same transient filtered expansion state, so it does not persist normal explorer expansion.

Validated:
- focused file explorer tests
- lint
- typecheck
- Electron file explorer flow for filtered collapse, drag-hover re-expand, and clearing the filter
